build.gradle

```diff
@@ -1,71 +1,115 @@
+    id "org.jetbrains.kotlin.jvm" version "1.3.31"
+    id "maven-publish"
 }
 
-apply plugin: 'kotlin'
-apply plugin: 'java'
-apply plugin: 'maven'
-apply plugin: 'idea'
-
 group = 'org.openbroker'
 version = '1.0-SNAPSHOT'
 description = 'Library for Open Broker'
 
-defaultTasks 'test'
+wrapper.gradleVersion = '5.4.1'
+
+ext {
+    kotlinLanguageVersion = '1.3'
+    jdkVersion = JavaVersion.VERSION_1_8
+
+    defaultEncoding = 'UTF-8'
 
-project.ext {
-    kotlin_version = '1.3.30'
-    kotlin_api_version = '1.3'
-    jvm_version = '1.8'
+    ver = [
+        'kotlin'     : '1.3.31',
+        'jackson'    : '2.9.9',
+        'junit'      : '5.4.2',
+        'cloudEvents': 'b8b9c6fef3',
+    ]
+
+    servicePackage = 'org.openbroker'
 }
 
-apply from: "jar.gradle"
+sourceCompatibility = jdkVersion
+targetCompatibility = jdkVersion
 
 repositories {
+    maven {
+        url "https://jitpack.io"
+        content {
+            includeGroupByRegex 'com\\.(github|gitlab|gitee).*'
+        }
+    }
     jcenter()
     mavenCentral()
-    maven { url "https://dl.bintray.com/kotlin/kotlinx" }
-    maven { url "https://jitpack.io" }
 }
 
 dependencies {
-    // Kotlin
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 
-    // Cloud Events
-    compile 'com.github.open-broker:cloud-events-jvm:b8b9c6fef37'
-    compile 'com.github.open-broker:cloud-events-jvm:b8b9c6fef37:sources'
+    implementation(
+        platform("org.jetbrains.kotlin:kotlin-bom:$ver.kotlin"),
+        platform("com.fasterxml.jackson:jackson-bom:$ver.jackson"),
+        "org.jetbrains.kotlin:kotlin-stdlib-jdk8",
+        "com.fasterxml.jackson.core:jackson-core",
+        "com.fasterxml.jackson.module:jackson-module-kotlin",
+        "com.fasterxml.jackson.datatype:jackson-datatype-jsr310",
+        "com.github.open-broker:cloud-events-jvm:$ver.cloudEvents",
+    )
 
-    // Jackson
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version:'2.9.8'
+    testImplementation(
+        platform("org.junit:junit-bom:$ver.junit"),
+        "org.junit.jupiter:junit-jupiter-api",
+    )
 
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.3.1")
-    testRuntime("org.junit.jupiter:junit-jupiter-engine:5.3.1")
+    testRuntime(
+        "org.junit.jupiter:junit-jupiter-engine",
+    )
 }
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
     kotlinOptions {
-        languageVersion = kotlin_api_version
-        apiVersion = kotlin_api_version
-        jvmTarget = jvm_version
-        javaParameters = true   // Useful for reflection.
+        javaParameters = true
+        jvmTarget = jdkVersion
+        freeCompilerArgs += ['-Xjsr305=strict', '-progressive']
```

Comment:
What does these compiler arguments do?

Reply:
About progressive argument: https://kotlinlang.org/docs/reference/whatsnew13.html#progressive-mode

About JSR-305: https://kotlinlang.org/docs/reference/java-interop.html#jsr-305-support. But in this case, i agree, jsr argument is redudant

Reply:
Okay, now we got some context on those arguments. I didn't question the changes in itself, I just wanted to know what they did and in which way they were an improvement for us.

I think they are good additionsl